netket/nn/symmetric_linear.py

```diff
@@ -713,8 +712,10 @@ def DenseSymm(symmetries, point_group=None, mode="auto", shape=None, **kwargs):
                 "in order to construct the space group"
             )
         sym = HashableArray(np.asarray(symmetries.automorphisms()))
-    elif isinstance(symmetries, PermutationGroup) or hasattr(symmetries, "__len__"):
+    elif isinstance(symmetries, (PermutationGroup, Array, Sequence)):
```

Comment:
While this does express our intention, it will unfortunately not work:
```python
>>> isinstance(a, netket.utils.types.Array)
TypeError: Subscripted generics cannot be used with class and instance checks
```
(This is because `Array` is a `typing.Union[...]` of several array-like types.) This was probably the original reason behind the use of `__len__` (we should leave a comment this time).

Maybe we should take a step back. What we want to accept, beside `Graph` and `PermutationGroup` which are easy to check for, are any array-like object that can be appropriately converted by `np.asarray` and `HashableArray`.

So maybe we should check for `HashableArray` first and then just put `sym = HashableArray(np.asarray(symmetries))` in the `else` branch? This would work unless anyone passes something completely incorrect (like a `str`), but that's not overly problematic (and common in Python code anyways).

Reply:
Passing a numpy array, a jax array or a list (or any non-hashable sequence) would still throw an error because of the non-hashable error induced by `HashablePartial` when applied to the `RBMSymm` model, though... I guess for the moment we must get either a `PermutationGroup`, a `Lattice`, or a `HashableArray`, and otherwise raise the error, right?

Reply:
For the moment I will only accept `HashableArray`, and not any other sequence, as there would be an error because of the hashing problem.

Reply:
Well, the error is actually independent from symmetry handling in DenseSymm. So I just did what you suggested.